airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java

```diff
@@ -268,22 +268,24 @@ String validatePrimaryKeys(final StreamId id,
 
     final String script = new StringSubstitutor(Map.of(
         "raw_table_id", id.rawTableId(QUOTE),
+        "raw_table_id_no_single_quotes", id.rawTableId(QUOTE).replaceAll("'", ""),
```

Comment:
`escapeSingleQuotedString` exists for this purpose! it also handles backslashes.

(... which is maybe a test case that the noCrash test should include :/ )